manifests/07-downloads-deployment.yaml

```diff
@@ -59,59 +59,42 @@ spec:
         - '-c'
         - |
           cat <<EOF >>/tmp/serve.py
-          import BaseHTTPServer, os, re, SimpleHTTPServer, socket, tarfile, tempfile, threading, time, zipfile
+          from BaseHTTPServer import HTTPServer, BaseHTTPRequestHandler
+          from SocketServer import ThreadingMixIn
+          import SimpleHTTPServer, os, re, signal, sys, tarfile, tempfile, zipfile
+          
+          class ThreadedHTTPServer(ThreadingMixIn, HTTPServer):
+              pass
 
-          # Launch multiple listeners as threads
-          class Thread(threading.Thread):
-              def __init__(self, i, socket):
-                  threading.Thread.__init__(self)
-                  self.i = i
-                  self.socket = socket
-                  self.daemon = True
-                  self.start()
-
-              def run(self):
-                  httpd = BaseHTTPServer.HTTPServer(addr, SimpleHTTPServer.SimpleHTTPRequestHandler, False)
-
-                  # Prevent the HTTP server from re-binding every handler.
-                  # https://stackoverflow.com/questions/46210672/
-                  httpd.socket = self.socket
-                  httpd.server_bind = self.server_close = lambda self: None
-
-                  httpd.serve_forever()
-
-          temp_dir = tempfile.mkdtemp()
-          print('serving from {}'.format(temp_dir))
-          os.chdir(temp_dir)
-          for arch in ['amd64']:
-              os.mkdir(arch)
-              for operating_system in ['linux', 'mac', 'windows']:
-                  os.mkdir(os.path.join(arch, operating_system))
-
-          for arch, operating_system, path in [
-                  ('amd64', 'linux', '/usr/bin/oc'),
-                  ('amd64', 'mac', '/usr/share/openshift/mac/oc'),
-                  ('amd64', 'windows', '/usr/share/openshift/windows/oc.exe'),
-                  ]:
-              basename = os.path.basename(path)
-              target_path = os.path.join(arch, operating_system, basename)
-              os.symlink(path, target_path)
-              base_root, _ = os.path.splitext(basename)
-              archive_path_root = os.path.join(arch, operating_system, base_root)
-              with tarfile.open('{}.tar'.format(archive_path_root), 'w') as tar:
-                  tar.add(path, basename)
-              with zipfile.ZipFile('{}.zip'.format(archive_path_root), 'w') as zip:
-                  zip.write(path, basename)
-
-          # Create socket
-          addr = ('', 8080)
-          sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-          sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
-          sock.bind(addr)
-          sock.listen(5)
-
-          [Thread(i, socket=sock) for i in range(100)]
-          time.sleep(9e9)
-          EOF
-          exec python2 /tmp/serve.py  # the cli image only has Python 2.7
+          signal.signal(signal.SIGTERM, lambda signum, frame: sys.exit(0))
+          
+          if __name__ == '__main__':
```

Comment:
Do we care about this?  We're going to execute the script immediately below; it will never be a library.  I'd rather drop this line and avoid the churny indent change.

Reply:
updated and rebased